hazardous/tests/test_metrics.py

```diff
@@ -0,0 +1,301 @@
+        ]
+    )
+
+    assert_array_almost_equal(loss, loss_expected)
+
+    ibs = integrated_brier_score(
+        y_train,
+        y_test,
+        y_pred,
+        times,
+        event_of_interest,
+    )
+
+    ibs_expected = 0.1257316251344779
+
+    assert abs(ibs - ibs_expected) < 1e-6
+
+
+@pytest.mark.parametrize("event_of_interest", [1, "any"])
+def test_brier_score_incidence_computer(event_of_interest):
+    times_, loss = brier_score(
+        y_train,
+        y_test,
+        y_pred,
+        times,
+        event_of_interest,
+    )
+    times_incidence_, loss_incidence = brier_score_incidence(
+        y_train,
+        y_test,
+        1 - y_pred,
+        times,
+        event_of_interest,
+    )
+
+    assert_array_equal(times_, times_incidence_)
+    assert_array_equal(loss, loss_incidence)
+
+    ibs = integrated_brier_score(
+        y_train,
+        y_test,
+        y_pred,
+        times,
+        event_of_interest,
+    )
+    ibs_incidence = integrated_brier_score_incidence(
+        y_train,
+        y_test,
+        1 - y_pred,
+        times,
+        event_of_interest,
+    )
+
+    assert ibs == ibs_incidence
+
+
+def test_brier_score_warnings_on_competive_event():
+    coef = np.random.choice([1, 2], size=y_train["event"].shape[0])
+    y_train["event"] *= coef
+
+    msg = "Computing the Brier Score only make sense"
+    with pytest.warns(match=msg):
+        brier_score(
+            y_train,
+            y_test,
+            y_pred,
+            times,
+            event_of_interest=2,
+        )
+
+    with pytest.warns(None):
+        brier_score_incidence(
+            y_train,
+            y_test,
+            1 - y_pred,
+            times,
+            event_of_interest=2,
+        )
+
+
+def test_brier_score_incidence_warnings_surv_input():
+    msg = "\n\nThe average shape of the y_pred curve is decreasing."
+    with pytest.warns(match=re.escape(msg)):
+        brier_score_incidence(
+            y_train,
+            y_test,
+            y_pred,
+            times,
+            event_of_interest="any",
+        )
+
+    with pytest.warns(None):
+        brier_score(
+            y_train,
+            y_test,
+            y_pred,
+            times,
+            event_of_interest="any",
+        )
+
+
+def test_brier_score_sampler():
+    sampler = BrierScoreSampler(y_train, random_state=0)
+    times_, y_binary, sample_weights = sampler.draw()
+
+    assert times_.shape == (y_train["event"].shape[0], 1)
+
+    y_binary_expected = np.array(
+        [0, 1, 1, 1, 1, 1, 1, 1, 1, 0],
+        dtype=np.int32,
+    )
+
+    assert_array_equal(y_binary[:10], y_binary_expected)
+
+    sample_weights_expected = np.array(
+        [
+            1.0892326,
+            1.02340426,
+            1.0141844,
+            1.05622555,
+            1.02340426,
+            1.0892326,
+            1.0141844,
+            1.05622555,
+            1.0892326,
+            1.03891038,
+        ]
+    )
```

Comment:
Here again, where do these value come from?

If this test were to fail, it wouldn't be clear if the test was wrong (and the new code was buggy) or if the test was correct and new version of the code introduced a regression. In the latter case, this test would not be very helpful to spot the root cause of the regression and that will make it tedious to debug.

It would be better to instead test on a very small data, e.g. with just 5 observations and put the expression of the computation of the expected as inline comments for each expected sample weight (with the KM formula).